test/services/samples/metadata/fields/update_service_test.rb

```diff
@@ -0,0 +1,157 @@
+                                         'value' => { 'value1' => 'value1' } } }
+
+          assert_authorized_to(:update_sample?, @sample32.project, with: ProjectPolicy,
+                                                                   context: { user: @user }) do
+            Samples::Metadata::Fields::UpdateService.new(@project29, @sample32, @user, params).execute
+          end
+        end
+
+        test 'update sample metadata without permission to update sample' do
+          user = users(:ryan_doe)
+          params = { 'update_field' => { 'key' => { 'metadatafield1' => 'metadatafield3' },
+                                         'value' => { 'value1' => 'value1' } } }
+
+          exception = assert_raises(ActionPolicy::Unauthorized) do
+            Samples::Metadata::Fields::UpdateService.new(@project29, @sample32, user, params).execute
+          end
+
+          assert_equal ProjectPolicy, exception.policy
+          assert_equal :update_sample?, exception.rule
+          assert exception.result.reasons.is_a?(::ActionPolicy::Policy::FailureReasons)
+          assert_equal I18n.t(:'action_policy.policy.project.update_sample?', name: @sample32.project.name),
+                       exception.result.message
+        end
+
+        test 'sample does not belong to project' do
+          params = { 'update_field' => { 'key' => { 'metadatafield1' => 'metadatafield3' },
+                                         'value' => { 'value1' => 'value1' } } }
+
+          Samples::Metadata::Fields::UpdateService.new(@project30, @sample32, @user, params).execute
+
+          assert @sample32.errors.full_messages.include?(
+            I18n.t('services.samples.metadata.fields.sample_does_not_belong_to_project', sample_name: @sample32.name,
+                                                                                         project_name: @project30.name)
+          )
+        end
+
+        test 'metadata was not changed' do
+          params = { 'update_field' => { 'key' => { 'metadatafield1' => 'metadatafield1' },
+                                         'value' => { 'value1' => 'value1' } } }
+
+          Samples::Metadata::Fields::UpdateService.new(@project29, @sample32, @user, params).execute
+
+          assert @sample32.errors.full_messages.include?(
+            I18n.t('services.samples.metadata.update_fields.metadata_was_not_changed')
+          )
+        end
+
+        test 'update metadata key to key that already exists' do
+          params = { 'update_field' => { 'key' => { 'metadatafield1' => 'metadatafield2' },
+                                         'value' => { 'value1' => 'value1' } } }
+
+          Samples::Metadata::Fields::UpdateService.new(@project29, @sample32, @user, params).execute
+
+          assert @sample32.errors.full_messages.include?(
+            I18n.t('services.samples.metadata.update_fields.key_exists', key: 'metadatafield2')
+          )
+        end
+      end
```

Comment:
Please add a test for a user not being able to update analysis provided metadata

Reply:
Updated in 2a79e5b24b98f16b80cf676792daa31801249850